portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -64,48 +63,64 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       response.getWriter().println(gson.toJson(comments));
     }
 
-    /*private String convertToJsonUsingGson(ArrayList comments){
-        Gson gson = new Gson();
-        String json = gson.toJson(comments);
-        return json;
-    }*/
+    //create a helper function that converts an Entity object into a Comment object and call it within the for loop
+    public Comments convertEntity(Entity entity){
+
+        String firstName = (String) entity.getProperty("firstName");
```

Comment:
all of these properties are repeated in multiple places and should be factored out.